python/ray/experimental/channel/torch_tensor_nccl_channel.py

```diff
@@ -50,31 +61,49 @@ def __init__(
         writes the worker-local
         ray.experimental.channel.serialization_context._SerializationContext
         when serializing data.
+
+        Args:
+            writer: The actor that may write to the channel. None signifies the
+                driver.
+            readers: The actors that may read from the channel. None signifies
+                the driver.
+            tensor_data_channel: A GPU-GPU channel for sending tensor data. Its
+                writer and readers should match the given writer and readers.
+            non_tensor_data_channel: A shared-memory channel for sending
+                non-tensor data. Its writer and readers should match the given
+                writer and readers.
+            static_non_tensor_data: Whether the non-tensor data sent through
+                this channel is static. For example, if all written values have
+                the form `{"my_tensor": torch.Tensor(...)}`, then this can be
```

Comment:
I am a bit confused. Why is `{"my_tensor": torch.Tensor(...)}` considered non-tensor data?

Reply:
I think it's because the tensors are within a dictionary, which is itself not a tensor. Since there are non-tensor components in the data, we can set `static_non_tensor_data=True` even though the dictionary contains tensors.

Reply:
Yes, that's correct.